pkg/controller/master/setup.go

```diff
@@ -65,15 +66,19 @@ func register(ctx context.Context, management *config.Management, options config
 func Setup(ctx context.Context, server *server.Server, controllers *server.Controllers, options config.Options) error {
 	scaled := config.ScaledWithContext(ctx)
 
+	startupComplete := make(chan bool, 1)
 	go leader.RunOrDie(ctx, "", "harvester-controllers", controllers.K8s, func(ctx context.Context) {
 		if err := register(ctx, scaled.Management, options); err != nil {
 			panic(err)
 		}
 		if err := scaled.Management.Start(options.Threadiness); err != nil {
 			panic(err)
 		}
-		<-ctx.Done()
+		startupComplete <- true
 	})
 
+	logrus.Debug("waiting for controller boot to complete")
+	<-startupComplete
+	logrus.Debug("controller boot complete")
```

Comment:
3 harvester pods case,  the pod that is not the leader will always be stuck here?

Reply:
yeah i realized that during testing. Since looking up helm chart is critical I have switched to using the client in addon controller rather than depending on caches.
The setup change has been rolled back.